servers/quarkus-auth/src/main/java/org/projectnessie/server/authn/NessieHttpAuthenticator.java

```diff
@@ -1,146 +0,0 @@
-      Event<AuthenticationSuccessEvent> authSuccessEvent,
-      BeanManager beanManager,
-      HttpBuildTimeConfig httpBuildTimeConfig,
-      Instance<HttpAuthenticationMechanism> httpAuthenticationMechanism,
-      Instance<IdentityProvider<?>> providers,
-      @ConfigProperty(name = "quarkus.security.events.enabled") boolean securityEventsEnabled) {
-    super(
-        identityProviderManager,
-        authFailureEvent,
-        authSuccessEvent,
-        beanManager,
-        httpBuildTimeConfig,
-        httpAuthenticationMechanism,
-        providers,
-        securityEventsEnabled);
-    this.base =
-        new BaseNessieHttpAuthenticator(
-            config,
-            () -> identityProviderManager.authenticate(AnonymousAuthenticationRequest.INSTANCE));
-  }
-
-  @Override
-  public Uni<SecurityIdentity> attemptAuthentication(RoutingContext context) {
-    if (!base.authEnabled) {
-      return base.anonymous();
-    }
-
-    return super.attemptAuthentication(context)
-        .onItem()
-        .transformToUni(
-            securityIdentity -> base.maybeTransform(securityIdentity, context.request().path()));
-  }
-
-  static class BaseNessieHttpAuthenticator {
-    private final boolean authEnabled;
-    private final Set<String> anonymousPaths;
-    private final Set<String> anonymousPathPrefixes;
-    private final Supplier<Uni<SecurityIdentity>> anonymousSupplier;
-
-    BaseNessieHttpAuthenticator(
-        QuarkusNessieAuthenticationConfig config,
-        Supplier<Uni<SecurityIdentity>> anonymousSupplier) {
-      this.authEnabled = config.enabled();
-      Set<String> paths = new HashSet<>();
-      config.anonymousPaths().ifPresent(paths::addAll);
-      config.anonymousPathPrefixes().ifPresent(paths::addAll);
-      this.anonymousPaths = paths;
-      this.anonymousPathPrefixes =
-          config
-              .anonymousPathPrefixes()
-              .map(prefixes -> prefixes.stream().map(p -> p + '/').collect(Collectors.toSet()))
-              .orElse(Set.of());
-      this.anonymousSupplier = anonymousSupplier;
-    }
-
-    Uni<SecurityIdentity> anonymous() {
-      return anonymousSupplier.get();
-    }
-
-    Uni<SecurityIdentity> maybeTransform(SecurityIdentity securityIdentity, String path) {
-      if (securityIdentity == null) {
-        if (path != null) {
-          // Allow certain preconfigured paths (e.g. health checks) to be serviced without
-          // authentication.
-          if (anonymousPaths.contains(path)
-              || anonymousPathPrefixes.stream().anyMatch(path::startsWith)) {
-            return anonymous();
-          }
-        }
-
-        // Disallow unauthenticated requests when requested by configuration.
-        // Note: Quarkus by default permits unauthenticated requests unless there are
-        // specific authorization rules that validate the security identity.
-        throw new AuthenticationFailedException("Missing or unrecognized credentials");
```

Comment:
Aren't we losing this use case with the new auth approach? The idea here was to deny access when the Nessie auth property is set in the config but a `SecurityIdentity` was not established. I'm not sure it works the same way with this PR.

Reply:
Unauthenticated requests fail with the changes in this PR

Reply:
right, my bad, I missed that, sorry.